src/App.tsx

```diff
@@ -9,36 +9,178 @@ import { PostDetails } from './components/PostDetails';
+    if (comments) {
+      setComments(comments.filter(comment => comment.id !== commendId));
+    }
+  };
+
+  useEffect(() => {
+    getUsers()
+      .then(usersFromServer => {
+        setUsers(usersFromServer);
+      })
+      .catch(() => {
+        setErrorMessage(Errors.UserApi);
+      });
+  }, []);
+
+  useEffect(() => {
+    if (selectedUser) {
+      setIsLoadingPosts(true);
+      setSelectedUsersPosts([]);
+      getPosts(selectedUserId)
+        .then(postsFromServer => {
+          setSelectedUsersPosts(postsFromServer);
+        })
+        .catch(() => {
+          setErrorMessage(Errors.PostApi);
+        })
+        .finally(() => {
+          setIsLoadingPosts(false);
+        });
+    }
+  }, [selectedUser]);
+
+  useEffect(() => {
+    if (selectedPost) {
+      setIsLoadingComments(true);
+
+      getComments(selectedPost.id)
+        .then(commentsFromServer => {
+          setComments(commentsFromServer);
+        })
+        .catch(() => {
+          setErrorMessage(Errors.CommentApi);
+        })
+        .finally(() => {
+          setIsLoadingComments(false);
+        });
+    }
+  }, [selectedPost]);
+
   return (
     <main className="section">
       <div className="container">
         <div className="tile is-ancestor">
           <div className="tile is-parent">
             <div className="tile is-child box is-success">
               <div className="block">
-                <UserSelector />
+                <UserSelector
+                  users={users}
+                  selectedUser={selectedUser}
+                  isDropDownActive={isDropDownActive}
+                  toggleDropDown={toggleDropDown}
+                  selectDropDownItem={selectDropDownItem}
+                  selectedUserId={selectedUserId}
+                  closeDropDown={closeDropDown}
+                />
               </div>
-
               <div className="block" data-cy="MainContent">
-                <p data-cy="NoSelectedUser">
-                  No user selected
-                </p>
+                {!selectedUser && (
+                  <p data-cy="NoSelectedUser">
+                    No user selected
+                  </p>
+                )}
 
-                <Loader />
+                {isLoadingPosts && <Loader />}
 
-                <div
-                  className="notification is-danger"
-                  data-cy="PostsLoadingError"
-                >
-                  Something went wrong!
-                </div>
+                {errorMessage === Errors.PostApi && (
+                  <div
+                    className="notification is-danger"
+                    data-cy="PostsLoadingError"
+                  >
+                    {Errors.PostApi}
+                  </div>
+                )}
 
-                <div className="notification is-warning" data-cy="NoPostsYet">
-                  No posts yet
-                </div>
+                {selectedUser && !errorMessage && (
```

Comment:
`selectedUser && !errorMessage` are better to be combined into a single variable